src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs

```diff
@@ -149,22 +161,12 @@ internal HResult BackupAndDeleteGlobalJson(string solutionDirectory, IVsSolution
                 pLogger.LogMessage((uint)__VSUL_ERRORLEVEL.VSUL_INFORMATIONAL, projectName, globalJson,
                     string.Format(VSResources.MigrationBackupFile, globalJson, globalJsonBackupPath));
                 _fileSystem.CopyFile(globalJson, globalJsonBackupPath, true);
-                try
-                {
-                    _fileSystem.RemoveFile(globalJson);
-
-                    var remover = new GlobalJsonRemover(_serviceProvider);
-                    HResult hr = solution.AdviseSolutionEvents(remover, out uint cookie);
-                    if (hr.Succeeded)
-                    {
-                        remover.SolutionCookie = cookie;
-                    }
-                }
-                catch (IOException e)
+
+                // Now parse the global.json, and remove the "sdk" element if it exists
+                var json = JsonConvert.DeserializeObject<JObject>(_fileSystem.ReadAllText(globalJson));
+                if (json.Remove("sdk"))
                 {
-                    pLogger.LogMessage((uint)__VSUL_ERRORLEVEL.VSUL_ERROR, projectName, globalJson,
-                        e.Message);
-                    return e.HResult;
+                    _fileSystem.WriteAllText(globalJson, JsonConvert.SerializeObject(json));
```

Comment:
>_fileSystem.WriteAllText(globalJson, JsonConvert.SerializeObject(json)); [](start = 20, length = 72)

Do you still want the try/catch around here so that if there are any IO issues they get reported to VS?